src/Geno/C++/Discord/DiscordRPC.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 Sebastian Kylander https://gaztin.com/
+ *
+ * This software is provided 'as-is', without any express or implied warranty. In no event will
+ * the authors be held liable for any damages arising from the use of this software.
+ *
+ * Permission is granted to anyone to use this software for any purpose, including commercial
+ * applications, and to alter it and redistribute it freely, subject to the following restrictions:
+ *
+ * 1. The origin of this software must not be misrepresented; you must not claim that you wrote the
+ *    original software. If you use this software in a product, an acknowledgment in the product
+ *    documentation would be appreciated but is not required.
+ * 2. Altered source versions must be plainly marked as such, and must not be misrepresented as
+ *    being the original software.
+ * 3. This notice may not be removed or altered from any source distribution.
+ */
+
+/*
+ * If you want to change this to your own discord app do the following:
+ * 1. Make the discord application in the discord website (https://discord.com/developers/applications/me).
+ * 2. Add the required art; C++ Logo, C# Logo, C Logo, And Geno Logo. You always add more such as a txt image.
+ * 3. Name the art with the correct names, C++ Logo would become "img_l_cpp_format", C Logo would be become "img_l_c_format", Geno Logo would be come "img_l_geno_wks_format" or "img_l_geno_prj_format" or "img_s_geno" etc.
+ * 4. Make sure that you set the default image to something!
+ */
+#define GENO_DISCORD_APP_ID 873985189037097052       // Geno IDE discord app
+#define GENO_DISCORD_APP_ID_STR "873985189037097052" // Geno IDE discord app
+
+#include "DiscordRPC.h"
+#include "Application.h"
+
+#include <iostream>
+#include <chrono>
+
+static const auto s_StartTime        = std::chrono::system_clock::now();
+static int64_t    s_StartInUnixTime  = -1;
```

Comment:
Why do we need static variables if this is a singleton class? Can't these simply be member variables?